CRM/Contribute/BAO/Contribution.php

```diff
@@ -1064,6 +1064,9 @@ protected static function getContributionTransactionInformation($contributionId,
       if ($paidByName === 'Check') {
         $val['check_number'] = $resultDAO->check_number;
       }
+      else {
+        $val['check_number'] = NULL;
```

Comment:
An alternative is that since resultDAO->check_number will always exist here just remove the surrounding `if`. But this works too.